### DISCUSSION TITLE
Is there a way to pass args to cmake build from the package.py?

### DISCUSSION THREAD

**Discussion by anderslanglands** [AcademySoftwareFoundation/rez#1343] [Q&A] [answered]
I've found `child_build_args` but that doesn't go to cmake (does it go to make, ninja etc?). I've hacked something in with:
```python
    build_args += getattr(package, "build_system_args", [])
```
in `create_build_system()` but I'm wondering if there's a proper way of doing it?

**Comment by nerdvegas** [comment, marked as answer]
You can just use `build_command` instead? It only has limited ability to
reference package attributes, but if you need more, you can make it an
early bound func (
https://github.com/AcademySoftwareFoundation/rez/wiki/Package-Definition-Guide#early-binding-functions
).

https://github.com/AcademySoftwareFoundation/rez/wiki/Package-Definition-Guide#build_command

Hth
A

**Comment by anderslanglands** [comment]
Thanks that makes sense. And if I make it a function, does that function
return a string to be executed by the shell or actually run the build using
eg subpocess.run()?

**Comment by nerdvegas** [comment]
The former. Most package attributes can be implemented as 'early'
functions, and they're expected to return the value that the standard
attribute would normally be (eg a list for 'requires', a string for
'name'). They literally get evaluated like that and the return value is
baked into the installed package.py.
